_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,428 @@
+---
+layout: post
+title:  "Reassigning partitions in Apache Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+As Apache Kafka users, we sometimes have to scale up/down the number of Kafka brokers in our cluster depending on the use case.
+Addition of extra brokers can be an advantage to handle massive load, and we can use Cruise Control for general rebalancing in Strimzi since it allows us to automate the balancing of load across the cluster but what if we are scaling down the clusters?
+Let us understand this with the help of an example, suppose there are certain number of brokers in a cluster and now we want to remove a broker from the cluster.
+We need to make sure that the broker which is going to be removed should not have any assigned partitions. Strimzi's integration of Cruise Control currently doesn't support doing this for you.
+You have to use some other tool to assign the partitions from the broker to be removed, to the remaining brokers.
+The most convenient tool for this job is the Kafka partition reassignment tool.
+
+<!--more-->
+
+##  Kafka reassignment partition tool
+
+The `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+While using the tool, you have to provide it with either of these two JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` that we can use directly or modify further.
+
+- The `reassignment.json` file is a configuration file that is used during the partition reassignment process. The reassignment partition tool will generate a proposal `reassignment.json` file based on a `topics.json` file. You can change the `reassignment.json` file as per your requirement and use it.
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+- You can reassign partitions between the brokers. For example, it can be used when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+- You can increase the number of partitions / replicas which can help with increasing the throughput of topics.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers often leads to additional interbroker network traffic, in addition to the normal traffic required for replication. To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
+This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes, for example `--throttle 5000000` sets the limit to 5 MB/s.
```

Comment:
```suggestion
This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes per second, for example `--throttle 5000000` sets the limit to 5 MB/s.
```